packages/eslint-plugin-react-hooks/__tests__/ESLintRuleExhaustiveDeps-test.js

```diff
@@ -3593,6 +3593,40 @@ const tests = {
         },
       ],
     },
+    {
+      code: normalizeIndent`
+        function MyComponent(props) {
+            function MyComponent(props) {
```

Comment:
We can remove one of the `MyComponent` functions from here and [here](https://github.com/facebook/react/pull/20247/files#diff-b7a2dc2bcc16791c0c4a8e1ef95755f0c1013eb93a5594df900499a1f1ad146bR3615-R3616)

Reply:
Good catch, thanks @reinaldosimoes.